app/scripts/controllers/preferences.test.js

```diff
@@ -274,21 +274,37 @@ describe('preferences controller', function () {
     });
   });
   describe('setUseTokenDetection', function () {
-    it('should default to false', function () {
+    it('should default to true', function () {
       const state = preferencesController.store.getState();
-      assert.equal(state.useTokenDetection, false);
+      if (process.env.TOKEN_DETECTION_V2) {
```

Comment:
Are these tests run multiple times with the flag on and then off?  If not, we should have two separate tests where the flag is shimmed on, and then off.